snapcraft/plugins/_python/_sitecustomize.py

```diff
@@ -0,0 +1,100 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import contextlib
+import glob
+import os
+from textwrap import dedent
+
+from ._python_finder import get_python_command
+from . import errors
+
+_SITECUSTOMIZE_TEMPLATE = dedent("""\
+    import site
+    import os
+
+    snap_dir = os.getenv("SNAP")
+    snapcraft_stage_dir = os.getenv("SNAPCRAFT_STAGE")
+    snapcraft_part_install = os.getenv("SNAPCRAFT_PART_INSTALL")
+
+    for d in (snap_dir, snapcraft_stage_dir, snapcraft_part_install):
+        if d:
+            site_dir = os.path.join(d, "{site_dir}")
+            site.addsitedir(site_dir)
+
+    if snap_dir:
+        site.ENABLE_USER_SITE = False""")
+
+
+def _get_user_site_dir(python_major_version, *, install_dir):
+    path_glob = os.path.join(
+        install_dir, 'lib', 'python{}*'.format(python_major_version),
+        'site-packages')
+    user_site_dirs = glob.glob(path_glob)
+    if not user_site_dirs:
+        raise errors.MissingUserSitePackagesError(path_glob)
+
+    return user_site_dirs[0][len(install_dir)+1:]
+
+
+def _get_sitecustomize_path(python_major_version, *, stage_dir, install_dir):
+    # Use the part's install_dir unless there's a python in the staging area
```

Comment:
fwiw, this could easily be transformed into a docsring by adding a _returns_ at the start of the sentence and a little grammar twitches here and there.

Reply:
I didn't deem it necessary since it was a private function, but I can do that if you like.

Reply:
I only propose it as it is almost the same